src/api/objects/ObjectAPI.js

```diff
@@ -365,6 +373,7 @@ export default class ObjectAPI {
                     savedReject = reject;
                 });
                 domainObject.persisted = persistedTime;
+                domainObject.createdBy = this.currentUsername;
```

Comment:
There's the potential for a race condition here. It's conceivable that the user would not have been resolved at this point if the user provider is slower than the object provider. Object save can happen automatically on load when migration scripts update objects from legacy models. I'd also prefer not to cache the logged in user anywhere but the user provider, and leave control of the caching strategy up to the user provider itself.

So, TL;DR, let's call the User API here and fetch the user from there each time. `save` is asynchronous so it's ok to do it here.

Reply:
sigh... duh. Thank you!